xchange-deribit/src/main/java/org/knowm/xchange/deribit/v2/service/DeribitAccountService.java

```diff
@@ -39,7 +38,22 @@ public AccountInfo getAccountInfo() throws IOException {
                   }
                 })
             .collect(Collectors.toList());
-    Wallet w = Wallet.Builder.from(balances).build();
-    return new AccountInfo(w);
+    Wallet wallet = Wallet.Builder.from(balances).build();
+
+    List<OpenPosition> openPositions =
+        currencies
+            .get()
+            .parallel()
```

Comment:
this is meant for operations without blocking IO

blocking a thread pool shared for all stream processing with HTTP calls is a very bad practice 

also IOException should not be wrapped in ExchangeException

please just do this in a loop

Reply:
Fixed both